app/src/main/java/org/simple/clinic/overdue/Appointment.kt

```diff
@@ -45,6 +49,42 @@ data class Appointment(
     class RoomTypeConverter : RoomEnumTypeConverter<Status>(Status::class.java)
   }
 
+  sealed class AppointmentType {
+
+    object Manual : AppointmentType()
+
+    object Automatic : AppointmentType()
+
+    data class Unknown(val actual: String) : AppointmentType()
+
+    object TypeAdapter : SafeEnumTypeAdapter<AppointmentType>(
+        knownMappings = mapOf(
+            Manual to "manual",
```

Comment:
Are there tests for checking if the conversion from Enum to string and string to enum is working as expected?

Reply:
This will be covered as part of the `AppointmentSyncAndroidTest`.

Whenever we add a new field to any of the payloads, we test syncing using the `*SyncAndroidTest` which internally calls the respective `SynceableRepository` methods to merge payload data to database models.
This will automatically get tested as part of that flow.